packages/nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -221,23 +337,20 @@ contract AssetHolder is IAssetHolder {
      * @notice Transfers as many funds escrowed against `channelId` as can be afforded for a specific destination. Assumes no repeated entries. Does not check allocationBytes against on chain storage.
      * @dev Transfers as many funds escrowed against `channelId` as can be afforded for a specific destination. Assumes no repeated entries. Does not check allocationBytes against on chain storage.
      * @param fromChannelId Unique identifier for state channel to transfer funds *from*.
-     * @param allocationBytes The abi.encode of AssetOutcome.Allocation
+     * @param allocation An AssetOutcome.AllocationItem[]
      * @param indices Array with each entry denoting the index of a destination to transfer funds to. Should be in increasing order. An empty array indicates "all".
      */
     function _transfer(
+        address asset,
         bytes32 fromChannelId,
-        bytes memory allocationBytes,
+        Outcome.AllocationItem[] memory allocation,
         uint256[] memory indices
-    ) internal {
-        Outcome.AllocationItem[] memory allocation = abi.decode(
-            allocationBytes,
-            (Outcome.AllocationItem[])
-        );
-        uint256 initialHoldings = holdings[fromChannelId];
+    ) internal returns (Outcome.AllocationItem[] memory) {
+        uint256 initialHoldings = holdings[asset][fromChannelId];
 
         (
-            Outcome.AllocationItem[] memory newAllocation,
-            bool safeToDelete,
+            Outcome.AllocationItem[] memory newAllocation, // safeToDelete
```

Comment:
Do you mean the next field is `safeToDelete`?

Reply:
Yes. I tried to put the comment on the next line, but prettier keeps moving it!

I will just remove the comment for now.